### PR TITLE
Glem restore bug fix

### PR DESCRIPTION
**Pull request by wangz10**
*Issue #, if available:*

*Description of changes:* Modify the `GLEM.restore_dense_model` and `GLEM.restore_sparse_model` such that it loads the correct layers for LM and GNN from either GLEM saved checkpoints or `GSgnnNodePredictionTrainer` saved checkpoints.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: python/graphstorm/config/argument.py [hunk 1]
```diff
@@ -816,16 +817,22 @@ def restore_model_layers(self):
...
-            return model_layers
-
-        return GRAPHSTORM_MODEL_ALL_LAYERS
+        if self.training_method["name"] == "glem":
+            # GLEM restore model layers conflicts with all layers
+            if model_layers == GRAPHSTORM_MODEL_ALL_LAYERS:
+                logging.warning("Restoring GLEM's LM from checkpoint doesn't support" \
+                                + " all layers, setting to: '%s'", GRAPHSTORM_MODEL_EMBED_LAYER
+                                )
+                model_layers = [GRAPHSTORM_MODEL_EMBED_LAYER]
```
Comment (zheng-da) [inline python/graphstorm/config/argument.py:834]: i'm not sure why we need this warning.

Comment (zheng-da) [reply python/graphstorm/config/argument.py:834]: for GLEM, do we need users to specify `restore_model_layers `?

Comment (wangz10) [reply python/graphstorm/config/argument.py:834]: Yes user will need to specify `restore_model_layers`. This is for loading a pre-trained LM, possible options are `['embed']` and `['embed', 'decoder']`. If not specified, we will default to `['embed']` and print this warning.

Comment (zheng-da) [reply python/graphstorm/config/argument.py:834]: Then the warning message should print the possible value of `restore_model_layers`

File: python/graphstorm/config/argument.py [hunk 2]
```diff
@@ -816,16 +817,28 @@ def restore_model_layers(self):
...
             model_layers = self._restore_model_layers.split(',')
             for layer in model_layers:
                 assert layer in GRAPHSTORM_MODEL_LAYER_OPTIONS, \
                     f"{layer} is not supported, must be any of {GRAPHSTORM_MODEL_LAYER_OPTIONS}"
-            return model_layers
-
-        return GRAPHSTORM_MODEL_ALL_LAYERS
+        # GLEM restore layers to the LM component, thus conflicting with all layers:
+        # use [GRAPHSTORM_MODEL_EMBED_LAYER, GRAPHSTORM_MODEL_DECODER_LAYER] to restore an LM
+        # checkpoint with decoder trained for node classification;
```
Comment (zheng-da) [inline python/graphstorm/config/argument.py:830]:
```suggestion
        # checkpoint with decoder trained for node classification.
        # For example, the check point is saved from a GLEM model.
```

**Review by zheng-da — APPROVED** [review]

File: python/graphstorm/model/node_glem.py [hunk 3]
```diff
@@ -108,11 +108,10 @@ def restore_dense_model(self, restore_model_path,
         self.lm.restore_dense_model(os.path.join(restore_model_path, 'LM'),
                                     model_layer_to_load)
         self.gnn.restore_dense_model(os.path.join(restore_model_path, 'GNN'),
-                                     model_layer_to_load)
+                                     ['gnn', 'decoder'])
 
     def restore_sparse_model(self, restore_model_path):
         self.lm.restore_sparse_model(os.path.join(restore_model_path, 'LM'))
-        self.gnn.restore_sparse_model(os.path.join(restore_model_path, 'GNN'))
```
Comment (zheng-da) [inline python/graphstorm/model/node_glem.py:115]: is this right?

Comment (zheng-da) [reply python/graphstorm/model/node_glem.py:115]: GNN model doesn't have sparse embeddings?

**Pull request merged**